src/vs/workbench/services/languageDetection/browser/languageDetectionSimpleWorker.ts

```diff
@@ -34,14 +34,15 @@ export class LanguageDetectionSimpleWorker extends EditorSimpleWorker {
 	private _modelOperations: ModelOperations | undefined;
 	private _loadFailed: boolean = false;
 
-	public async detectLanguage(uri: string, langBiases: Record<string, number> | undefined, preferHistory: boolean): Promise<string | undefined> {
+	public async detectLanguage(uri: string, langBiases: Record<string, number> | undefined, preferHistory: boolean, supportedLangs?: string[]): Promise<string | undefined> {
 		const languages: string[] = [];
 		const confidences: number[] = [];
 		const stopWatch = new StopWatch(true);
 		const documentTextSample = this.getTextForDetection(uri);
 		if (!documentTextSample) { return; }
 
 		const neuralResolver = async () => {
+			if (supportedLangs?.length) { return undefined; /* neural resolver doesnt support language tuning */ }
```

Comment:
Could you just check if the `language` in the next line is a `supportedLang` before pushing to languages & confidence?

Reply:
Makes sense 👍 do the confidences need to be scaled differently?

Reply:
Actually sorry, line 52 ... we only return 1 language.... Can you just return the first language that's in `supportedLang`? The `languages are already sorted by confidence by the `detectLanguagesImpl`.